ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/update/UpdatePathSpec.scala

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.update
+
+import org.scalatest.EitherValues
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+
+class UpdatePathSpec extends AnyFreeSpec with Matchers with EitherValues {
+
+  "parse valid update paths" in {
+    UpdatePath
+      .parseAll(
+        Seq(
+          "foo.bar",
+          "foo",
+          "foo.bar!merge",
+          "foo.bar!replace",
+        )
+      )
+      .value shouldBe Seq(
+      UpdatePath(List("foo", "bar"), UpdatePathModifier.NoModifier),
+      UpdatePath(List("foo"), UpdatePathModifier.NoModifier),
+      UpdatePath(List("foo", "bar"), UpdatePathModifier.Merge),
+      UpdatePath(List("foo", "bar"), UpdatePathModifier.Replace),
+    )
+  }
+
+  "raise errors when parsing invalid paths" in {
```

Comment:
you can also have two bangs or a bang followed by nothing

Reply:
added test cases